### PR TITLE
iCloud Photo Library Downloading

### PR DESCRIPTION
**Pull request by jlalvarez18**
- added getMediaFilesBetween() to fetch between dates
- updated example to work with latest XCode and get files for the month
- introduced MediaAsset

### CODE REVIEW

File: example/.flutter-plugins-dependencies [hunk 1]
```diff
@@ -0,0 +1 @@
+{"info":"This is a generated file; do not edit or check into version control.","plugins":{"ios":[{"name":"permission_handler","path":"/Users/juan/dev/flutter/.pub-cache/hosted/pub.dartlang.org/permission_handler-3.1.0/","dependencies":[]},{"name":"media_picker_builder","path":"/Users/juan/private_repos/media_picker_builder/","dependencies":[]}],"android":[{"name":"permission_handler","path":"/Users/juan/dev/flutter/.pub-cache/hosted/pub.dartlang.org/permission_handler-3.1.0/","dependencies":[]},{"name":"media_picker_builder","path":"/Users/juan/private_repos/media_picker_builder/","dependencies":[]}],"macos":[],"linux":[],"windows":[],"web":[]},"dependencyGraph":[{"name":"permission_handler","dependencies":[]},{"name":"media_picker_builder","dependencies":[]}],"date_created":"2020-10-15 17:59:47.554714","version":"1.22.0"}
```
Comment (Sauraus) [inline example/.flutter-plugins-dependencies]: Please commit a proper .gitignore generated at https://gitignore.io

File: example/ios/Flutter/.last_build_id [hunk 2]
```diff
@@ -0,0 +1 @@
+19ac9e0be5b5e961faee53688bb5943d
```
Comment (Sauraus) [inline example/ios/Flutter/.last_build_id]: Not sure this file belongs in git.

**Review by Sauraus — CHANGES_REQUESTED** [review]
@jlalvarez18 lets do a call with @qbait, @adamstyrc and myself on Monday to walk,through these changes, there is a lot here.

File: .gitignore [hunk 3]
```diff
@@ -1,7 +1,62 @@
-.DS_Store
-.dart_tool/
+# Created by https://www.toptal.com/developers/gitignore/api/flutter
+# Edit at https://www.toptal.com/developers/gitignore?templates=flutter
```
Comment (Sauraus) [inline .gitignore]: We probably want this to be 

https://www.toptal.com/developers/gitignore?templates=dart,swift,kotlin,flutter,intellij,androidstudio,visualstudiocode,osx,windows

**Review by Sauraus — CHANGES_REQUESTED** [review]
What about the albumname that @qbait removed entirely from the code base, did you come across that and why it was giving us grief?

File: .gitignore (hunk 3 above)
Comment (jlalvarez18) [reply .gitignore]: Nice!

**Comment by jlalvarez18** [comment]
@Sauraus album name? My new re-write doesn't care about albums. You fetch based on type and date range. The old codebase used albums, then iterated through the albums, then fetched files from the cloud whether you liked it or not, then returned you the results.

**Review by Sauraus — APPROVED** [review]

**Comment by Sauraus** [comment]
> @Sauraus album name? My new re-write doesn't care about albums. You fetch based on type and date range. The old codebase used albums, then iterated through the albums, then fetched files from the cloud whether you liked it or not, then returned you the results.

So what if I want to import from an album that's no longer possible?

**Pull request merged**